src/AsyncBridge/Runtime.CompilerServices/TaskAwaiter.cs

```diff
@@ -254,17 +254,27 @@ private static void RunNoException(Action continuation)
         ///   Copies the exception's stack trace so its stack trace isn't overwritten.
         /// </summary>
         /// <param name="exc"> The exception to prepare. </param>
+        [SecuritySafeCritical]
         internal static Exception PrepareExceptionForRethrow(Exception exc)
         {
             if (s_prepForRemoting != null)
             {
                 try
                 {
-                    s_prepForRemoting.Invoke(exc, s_emptyParams);
+#if !PORTABLE
+                    new PermissionSet(Security.Permissions.PermissionState.Unrestricted).Assert();
+#endif
+                    return (Exception)s_prepForRemoting.Invoke(exc, s_emptyParams);
                 }
                 catch
                 {
                 }
+#if !PORTABLE
+                finally
+                {
+                    CodeAccessPermission.RevertAssert();
+                }
+#endif
```

Comment:
Can we have a test that fails when the Assert() is removed?
Also, I noticed that `s_prepForRemoting.Invoke` became `return (Exception)`. Is that fixing a separate bug? Can we have a test that fails without the `return`?

Reply:
For the return, the PrepareRemoting method returns the exception. It just happens to return 'this' so ignoring it doesn't cause a problem. Personal preference to follow the obvious desire of the original code and use the return value.

The tests don't fail, but in partial trust they will lose their full stack trace. I wasn't sure how to write a reliable test for that without some crazy string checking that would also make the test localisation specific.

Reply:
That could work! I'll give it a shot.